spec/controllers/users/invitations_controller_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Users::InvitationsController, type: :controller do
+  let(:family) { create(:family) }
+  let(:user) { create(:user) }
+  let(:token) { JsonWebToken.encode(user_id: user.id) }
+  let(:params) do
+    {
+      user: {
+        first_name: Faker::Name.first_name,
+        last_name: Faker::Name.last_name,
+        phone: Faker::Base.numerify('###-###-####'),
+        email: Faker::Internet.email,
+        avatar: Rack::Test::UploadedFile.new(ENV['LOCAL_IMAGE_PATH']),
+        address: Faker::Address.full_address,
+        member_type: User.member_types.keys.sample,
+        family_id: user.family.id,
+        password: 'password',
+        password_confirmation: 'password'
+      }
+    }
+  end
+  let(:update_params) do
+    {
+      user: {
+        first_name: Faker::Name.first_name,
+        last_name: Faker::Name.last_name,
+        phone: Faker::Base.numerify('###-###-####'),
+        email: Faker::Internet.email,
+        avatar: Rack::Test::UploadedFile.new(ENV['LOCAL_IMAGE_PATH']),
+        address: Faker::Address.full_address,
+        member_type: User.member_types.keys.sample,
+        family_id: user.family.id,
+        password: 'password',
+        password_confirmation: 'password',
+        invitation_token: invitation_token
+      }
+    }
+  end
+  let(:invitation_token) { response.parsed_body['data']['invite_token'] }
+  let(:send_post_request) { post :create,
+                            params: params.merge(current_user: user),
+                            as: :json }
+  let(:send_put_request) { put :update,
+                           params: update_params.merge(current_user: user),
+                           as: :json }
```

Comment:
to test update action i need send POST request first, to parse "invite_token" from his response and then to paste it to PUT request. so i need to send 2 requests at the same time
```
    before do
      send_post_request
      send_put_request
    end
```

Reply:
You don't need to do a request to get token. See: [jwt testing](https://github.com/waiting-for-dev/devise-jwt#testing). It will looks like:
```ruby
  let(:headers) { { 'Accept' => 'application/json', 'Content-Type' => 'application/json' } }
  let(:auth_headers) do
    Devise::JWT::TestHelpers.auth_headers(headers, user)
  end

  before { request.headers.merge!(auth_headers) }
```

Reply:
i`m sorry, but i didnt understand your example with JWT, i made factory with 'invitation_token' and sent request with encrypted token. it works.